installer/Installer/Model.cs

```diff
@@ -103,60 +105,99 @@ class Program
         static string ProgramName = "THUAI6";                     // 要运行或下载的程序名称
         static string playerFolder = "player";                    // 选手代码保存文件夹路径
         static string startName = "maintest.exe";                 // 启动的程序名
+        static int filenum = 0;                                   // 总文件个数
 
         public class Data
         {
-            public static string path = "";      // 标记路径记录文件THUAI6.dat的路径
+            public static string path = "";      // 标记路径记录文件THUAI6.json的路径
             public static string FilePath = "";  // 最后一级为THUAI6文件夹所在目录
             public static string dataPath = "";  // C盘的文档文件夹
             public Data(string path)
             {
                 // dataPath = Environment.GetFolderPath(Environment.SpecialFolder.MyDocuments);
                 dataPath = new DirectoryInfo(".").FullName;
-                Data.path = System.IO.Path.Combine(dataPath, "THUAI6.dat");
+                Data.path = System.IO.Path.Combine(dataPath, "THUAI6.json");
                 if (File.Exists(Data.path))
                 {
+                    var dict = new Dictionary<string, string>();
                     using (StreamReader r = new StreamReader(Data.path))
-                        FilePath = r.ReadLine().Replace('\\', '/');  // 读取THUAI6.dat文件的第一行
+                    {
+                        string json = r.ReadToEnd();
+                        if (json == null || json == "")
+                        {
+                            json += @"{""THUAI6""" + ":" + @"""2023""}";
+                        }
+                        dict = JsonConvert.DeserializeObject<Dictionary<string, string>>(json);
+                        if (dict != null && dict.ContainsKey("installpath"))
+                        {
+                            FilePath = dict["installpath"].Replace('\\', '/');
+                        }  //读取安装路径
+                    }
+                    dict?.TryAdd("installpath", @path);
+                    FileStream fs = new FileStream(Data.path, FileMode.Create, FileAccess.ReadWrite);
+                    StreamWriter sw = new StreamWriter(fs);
```

Comment:
```suggestion
                    using StreamWriter sw = new StreamWriter(fs);
```
用 `using` 可以不用之后的 `Flush` 和 `Close`